### PR TITLE
font dropdown + settings modal cleanup + rustfmt

### PR DESCRIPTION
**Pull request by SpanishPear**
doesn't actually change font size currently but UI works

this PR comes also with a lot of cleaning up of the settings modal and some new components (Heading, Dropdown, toggle)

### CODE REVIEW

**Comment by SpanishPear** [comment]
probably should have done the rustfmt in a different pr
owell

File: crates/mipsy_web/src/components/settings_modal.rs [hunk 1]
```diff
@@ -36,152 +41,56 @@ pub fn render_modal(props: &ModalProps) -> Html {
...
-                                        "Opt out"
-                                    }
+                    <div class="w-9/12">
+                        <Heading
+                            title="Editor Font Size"
+                            subtitle="Adjust the font size of the editor"
+                        />
+
+                        <div class="w-3/12">
+                            <Dropdown
```
Comment (SpanishPear) [inline crates/mipsy_web/src/components/settings_modal.rs:57]: this is the font dropdown actual impl :)

File: crates/mipsy_web/src/components/data_segment.rs [hunk 2]
```diff
@@ -40,7 +40,8 @@ pub fn data_segment(props: &DataSegmentProps) -> Html {
         .map(|(key, val)| (key.clone(), val.clone()))
         .collect::<Vec<_>>();
 
-    let registers = Some(props.state.mips_state.clone()).clone()
+    let registers = Some(props.state.mips_state.clone())
+        .clone()
```
Comment (abiramen) [inline crates/mipsy_web/src/components/data_segment.rs:44]: clone++ :yum:

**Review by abiramen — APPROVED** [review]
good content!

**Pull request merged**